udata_datalake_service/background_tasks.py

```diff
@@ -39,33 +46,46 @@ def save_resource_to_minio(resource_file, key, resource):
     )
     try:
         with open(resource_file.name, "rb") as f:
-            s3.upload_fileobj(f, os.getenv("MINIO_BUCKET"), MINIO_FOLDER + key + "/" + resource["id"])
+            s3.upload_fileobj(f, os.getenv("MINIO_BUCKET"), MINIO_FOLDER + "/" + key + "/" + resource["id"])
         logging.info(
-            "Resource saved into minio at {}".format(
-                os.getenv("MINIO_URL")
-                + os.getenv("MINIO_BUCKET")
-                + MINIO_FOLDER
-                + key
-                + "/"
-                + resource["id"]
-            )
+            f"Resource saved into minio at {get_resource_minio_url(key, resource)}"
         )
     except ClientError as e:
         logging.error(e)
 
 
 @celery.task
-def manage_resource(key, resource):
+def manage_resource(dataset_id: str, resource: dict):
     logging.info(
-        "Processing task for resource {} in dataset {}".format(resource["id"], key)
+        "Processing task for resource {} in dataset {}".format(resource["id"], dataset_id)
     )
     try:
         tmp_file = download_resource(resource["url"])
-        save_resource_to_minio(tmp_file, key, resource)
+        
+        # Check resource MIME type
+        mime_type = magic.from_file(tmp_file.name, mime=True)
+        if mime_type in ["text/plain", "text/csv"]:
+            # Save resource only if CSV
+            try:
+                # Raise ValueError if file is not a CSV
+                agate.Table.from_csv(tmp_file.name, sniff_limit=4096, row_limit=40)
+                save_resource_to_minio(tmp_file, dataset_id, resource)
+                storage_location = {"netloc": os.getenv("MINIO_URL"), "bucket": os.getenv("MINIO_BUCKET"), "key": MINIO_FOLDER + "/" + dataset_id + "/" + resource["id"]}
```

Comment:
can we add in `resource.stored` message the `mime_type` of resource. It could be useful if we decide later to download other kind of files (for instance Excel files) but not process them for further analysis.